utils.c

```diff
@@ -108,34 +110,31 @@ int list_dir(connection_t *connection_prop, char *html, unsigned int bufsize, bo
             //Scaling the file's size
             unsigned long long int size = f_prop.st_size;
             if (size < 1024) {
-                measure="B";
+                measure=" B";
             } else if ((size = (size / 1024)) < 1024) {
-                measure="KiB";
+                measure=" KiB";
             } else if ((size = (size / 1024)) < 1024) {
-                measure="MiB";
+                measure=" MiB";
             } else {
                 size = size / 1024;
-                measure="GiB";
+                measure=" GiB";
             }
 
-            if (i % 2 == 0)
-                color = "white";
-            else
-                color = "#EAEAEA";
-
             printf_s=snprintf(html+pagesize,maxsize,
                               "<tr style=\"background-color: %s;\"><td>f</td><td><a href=\"%s\">%s</a></td><td>%lld%s</td><td>%s</td></tr>\n",
                               color, namelist[i]->d_name, namelist[i]->d_name, (long long int)size, measure,last_modified);
             maxsize-=printf_s;
             pagesize+=printf_s;
+            color = (color == "white" ? "#E0E0E0" : "white");
 
         } else if (S_ISDIR(f_mode)) { //Directory entry
             //Table row for the dir
             printf_s=snprintf(html+pagesize,maxsize,
-                              "<tr style=\"background-color: #DFDFDF;\"><td>d</td><td><a href=\"%s/\">%s/</a></td><td>-</td><td>%s</td></tr>\n",
-                              namelist[i]->d_name, namelist[i]->d_name,last_modified);
+                              "<tr style=\"background-color: %s;\"><td><b>d</b></td><td><a href=\"%s\"><b>%s/</b></a></td><td></td><td>%s</td></tr>\n",
+                              color, namelist[i]->d_name, namelist[i]->d_name,last_modified);
             maxsize-=printf_s;
             pagesize+=printf_s;
+            color = (color == "white" ? "#E0E0E0" : "white");
```

Comment:
Excellent! Then I will just proceed to customize the code for my purpose for now, please ignore further changes. If I find anything worthy of inclusion for you, I'll do it on a fresh rebase.